lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -704,6 +704,23 @@ tokenize(String, Line, Column, OriginalScope, Tokens) ->
 previous_was_dot([{'.', _} | _]) -> true;
 previous_was_dot(_) -> false.
 
+unexpected_identifier_token([T | Rest] = String, Line, Column, Scope, Tokens) ->
+  AsNFKC = unicode:characters_to_nfkc_list(unicode:characters_to_nfc_list([T])),
+  case (Scope#elixir_tokenizer.identifier_tokenizer):tokenize(AsNFKC) of
+    {error, _} ->
+      unexpected_token(String, Line, Column, Scope, Tokens);
+    _NFKCWouldWork ->
+      Wrong = [T],
+      WrongCodepoints = list_to_codepoint_hex([T]),
+      NFKCCodepoints = list_to_codepoint_hex(AsNFKC),
+      Message = io_lib:format("Unicode codepoint did not parse, but its NFKC normalization would parse.\n\n"
+                              "  Got:  ~ts (codepoints~ts) at column ~p\n"
+                              "  NFKC: ~ts (codepoints~ts)\n\n"
+                              "Syntax error at:",
+                              [Wrong, WrongCodepoints, Column, AsNFKC, NFKCCodepoints]),
```

Comment:
I am thinking about improving the error message to not mention NKFC. I would also show the whole word, as we did for the previous NFC error, because the user can simply copy and paste it. If someone really cares about codeopints, they can compare it:

```
** (SyntaxError) iex:4:1: Elixir expects unquoted Unicode atoms, variables, and calls to be in NFC form.

Got:

    "foo𝚳" (code points 0x0066 ...)

HINT: you can rewrite the term above to a canonically equivalent format that is accepted by Elixir:

    "fooΜ" (code points 0x0066 ...)

Syntax error before: fooM
```

WDYT?

Reply:
*nods* yeah, that would be better; testing switching some logic in continue() in tokenizer.ex now to support that

Reply:
Huh! So there ended up being a rabbit hole there.

The example above normalizes to a Greek M, not a Latin one -- which would make the whole thing a mixed-script token, (no intersection between {Latin}, {Greek}), which causes it to fail to tokenize -- so the NFKC normalized version of the full token wouldn't be suggested.

But when thinking about how to fix that, I thought, wait -- 💡 **what about our confusability skeletons ☠️ from the Tokenizer.Security module**?

So I have it try both, one after the other, and include those suggestions after mixed-script warnings, too, when that would fix it!

- [test with comment mentioning how it works](https://github.com/elixir-lang/elixir/blob/21dc8a86ed6c7ee549555c1ff13cfdfe1436050c/lib/elixir/test/elixir/kernel/parser_test.exs#L630-L679)
- [updated mixed-script test with suggestion](https://github.com/elixir-lang/elixir/blob/21dc8a86ed6c7ee549555c1ff13cfdfe1436050c/lib/elixir/test/elixir/kernel/warning_test.exs#L72-L89)

Most of the code handling that ^^^ has been added to `elixir_tokenizer.erl`